src/planet-a/SubdivSkeleton.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2021 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "SubdivSkeleton.h"
+
+using namespace planeta;
+
+SkTriGroupId SubdivTriangleSkeleton::tri_subdiv(SkTriId triId,
+                                                std::array<SkVrtxId, 3> vrtxMid)
+{
+    SkeletonTriangle &rTri = tri_at(triId);
+
+    if (rTri.m_children.has_value())
+    {
+        throw std::runtime_error("SkeletonTriangle is already subdivided");
```

Comment:
Is this the right thing to do?

If the triangle is already subdivided, but the function gets called a second time, replacing the previous work with the new result should leave you with the same outcome. So all that's wasted is some repeated work.

Throwing an exception, however, doesn't address the function being called twice. That should be handled at a higher level, i think.

Reply:
This function shouldn't be responsible for deleting the previous value if it existed. That would add uncertainty, not convenience.